perl/content.md

```diff
@@ -32,6 +32,32 @@ For many simple, single file projects, you may find it inconvenient to write a c
 $ docker run -it --rm --name my-running-script -v "$PWD":/usr/src/myapp -w /usr/src/myapp %%IMAGE%%:5.20 perl your-daemon-or-script.pl
 ```
 
+## Signal handling behavior notice
+
+As Perl will run as PID 1 by default in containers (unless an [ENTRYPOINT](https://docs.docker.com/engine/reference/builder/#entrypoint) is set,) special care needs to be considered when expecting to send signals (particularly SIGINT or SIGTERM) to it. For example, running
+
+```console
+$ docker run -it --name kill-me-plz --rm %%IMAGE%%:5.20 perl -E 'sleep 300'
+```
+
+and doing on another terminal,
+
+```console
+$ docker exec kill-me-plz kill 1
+```
+
+will *not* stop the perl running on the `kill-me-plz` container (it will keep running until the `sleep 300` finishes.) To do so, one must set a signal handler like this:
+
+```console
+$ docker run -it --name kill-me-now --rm %%IMAGE%%:5.20 perl -E '$SIG{TERM} = sub { say "recv TERM" }; sleep 300'
+```
+
+so doing `docker exec kill-me-now kill 1` (or the simpler `docker stop kill-me-now`) will immediately stop the container, and print `recv TERM` in the other terminal.
```

Comment:
Just two mior comments: maybe not name the containers about killing, I don't want to unintentionally upset or hurt someone; use `docker kill -s TERM some-perl` instead of `docker exec kill-me-now kill 1`.

Reply:
Thanks, updated with more benign names :muscle: